packages/core-mobile/app/services/walletconnectv2/WalletConnectService.ts

```diff
@@ -322,6 +327,156 @@ class WalletConnectService implements WalletConnectServiceInterface {
+  }
+
+  updateSessionForNonEvmAccount = async ({
+    session,
+    account,
+    isTestnet = false
+  }: {
+    session: SessionTypes.Struct
+    account: CorePrimaryAccount
+    isTestnet?: boolean
+  }): Promise<void> => {
+    const topic = session.topic
+    Logger.info(
+      `updating WC session '${session.peer.metadata.name}' with non evm chains.`
+    )
+
+    const namespaces: SessionTypes.Namespaces = {}
+
+    for (const key of Object.keys(session.namespaces)) {
+      const namespace = session.namespaces[key]
+
+      if (!namespace) continue
+
+      // update chain and account lists for avax and bip122 namespaces
+      // when we toggle between testnet and mainnet, or change active account
+      if (key === BlockchainNamespace.AVAX) {
+        const caip2ChainIds = isTestnet
+          ? [
+              AvalancheCaip2ChainId.C_TESTNET,
+              AvalancheCaip2ChainId.P_TESTNET,
+              AvalancheCaip2ChainId.X_TESTNET
+            ]
+          : [
+              AvalancheCaip2ChainId.C,
+              AvalancheCaip2ChainId.P,
+              AvalancheCaip2ChainId.X
+            ]
+
+        this.updateNamespaceForNonEvmCaip2ChainId({
+          account,
+          namespace,
+          caip2ChainIds,
+          blockchainNamespace: BlockchainNamespace.AVAX
+        })
+      }
+
+      if (key === BlockchainNamespace.BIP122) {
+        const caip2ChainIds = isTestnet
+          ? [BitcoinCaip2ChainId.TESTNET]
+          : [BitcoinCaip2ChainId.MAINNET]
+
+        this.updateNamespaceForNonEvmCaip2ChainId({
+          account,
+          namespace,
+          caip2ChainIds,
+          blockchainNamespace: BlockchainNamespace.BIP122
+        })
+      }
+      namespaces[key] = { ...namespace }
+    }
+
+    // check if dapp is online first
+    await this.client.engine.signClient.ping({ topic })
+
+    await this.client.updateSession({
+      topic,
+      namespaces
+    })
+  }
+
+  private updateNamespaceForNonEvmCaip2ChainId = ({
+    account,
+    namespace,
+    caip2ChainIds,
+    blockchainNamespace
+  }: {
+    account: CorePrimaryAccount
+    namespace: SessionTypes.Namespace
+    caip2ChainIds: string[]
+    blockchainNamespace: BlockchainNamespace
+  }): void => {
+    caip2ChainIds.forEach(caip2ChainId => {
+      updateChainListInNamespace({ chains: namespace.chains, caip2ChainId })
+      const addressWithCaip2ChainId = getAddressWithCaip2ChainId({
+        account,
+        blockchainNamespace,
+        caip2ChainId
+      })
+      addressWithCaip2ChainId &&
+        updateAccountListInNamespace({
+          account: addressWithCaip2ChainId,
+          accounts: namespace.accounts
+        })
+    })
+  }
+
+  private updateSessionWithTimeoutForNonEvm = async ({
+    session,
+    account,
+    isTestnet
+  }: {
+    session: SessionTypes.Struct
+    account: CorePrimaryAccount
+    isTestnet?: boolean
+  }): Promise<void> => {
+    const isCoreApp = isCoreDomain(session.peer.metadata.url)
+
```

Comment:
this is important to have but I don't think we should have core specific logic in this service. we should move it out in the future.